pkg/controller/imagemap_controller.go

```diff
@@ -0,0 +1,66 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ir "github.com/pivotal/image-relocation/pkg/api/v1alpha1"
+	webhookv1alpha1 "github.com/pivotal/image-relocation/pkg/api/v1alpha1"
+	"github.com/pivotal/image-relocation/pkg/multimap"
+)
+
+// ImageMapReconciler reconciles a ImageMap object
+type ImageMapReconciler struct {
+	client.Client
+	Log logr.Logger
+	Map multimap.Composite
+}
+
+func (r *ImageMapReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("ImageMap", req.NamespacedName)
+
+	var imageMap ir.ImageMap
+	if err := r.Get(ctx, req.NamespacedName, &imageMap); err != nil {
+		if apierrs.IsNotFound(err) {
+			log.Info("deleting")
+			_ = r.Map.Delete(req.NamespacedName.String()) // ignore error in case it has already been deleted
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "unable to get ImageMap")
+		return ctrl.Result{}, err
+	}
+
+	log.Info("adding")
+	if err := r.Map.Add(req.NamespacedName.String(), imageMap.Spec.Map); err != nil {
```

Comment:
if this is a cluster scoped resource, then there will never be a namespace, you can just use name.

Reply:
I'd like to keep it as it is in case someone uses a non-cluster scoped configuration and then they'll at least get sane results.